x-pack/test/security_solution_api_integration/test_suites/detections_response/rules_management/rule_bulk_actions/trial_license_complete_tier/perform_bulk_action.ts

```diff
@@ -2651,102 +2653,225 @@ export default ({ getService }: FtrProviderContext): void => {
             'default'
           );
 
-          const { gapEvents } = await generateGapsForRule(es, rule, 100);
-          generatedGapEvents[rule.id] = {
-            rule,
-            gapEvents: gapEvents.map((gapEvent) => {
-              if (!gapEvent._id) {
-                throw new Error('generated gap event id cannot be undefined');
-              }
-              return { ...gapEvent.kibana.alert.rule.gap, _id: gapEvent._id };
-            }),
-          };
+          createdRules.push(rule);
+
+          backfillEnd = new Date();
+          backfillStart = new Date(backfillEnd.getTime() - 24 * 60 * 60 * 1000);
         }
+
+        createdRuleIds = createdRules.map(({ id }) => id);
       });
 
-      it('should trigger the backfilling of the gaps for the rules in the request', async () => {
-        // Only backfill the first 2 rules
-        const ruleIdsToBackfill = Object.keys(generatedGapEvents).slice(0, 2);
+      describe('scheduling gap fills for rules', () => {
+        beforeEach(async () => {
+          generatedGapEvents = {};
+          for (const rule of createdRules) {
+            const { gapEvents } = await generateGapsForRule(es, rule, 100);
+            generatedGapEvents[rule.id] = {
+              rule,
+              gapEvents: gapEvents.map((gapEvent) => {
+                if (!gapEvent._id) {
+                  throw new Error('generated gap event id cannot be undefined');
+                }
+                return { ...gapEvent.kibana.alert.rule.gap, _id: gapEvent._id };
+              }),
+            };
+          }
 
-        // Trigger the backfill for the selected rules
-        const { body } = await securitySolutionApi
-          .performRulesBulkAction({
-            query: {},
-            body: {
-              ids: ruleIdsToBackfill,
-              action: BulkActionTypeEnum.fill_gaps,
-              [BulkActionTypeEnum.fill_gaps]: {
-                start_date: backfillStart.toISOString(),
-                end_date: backfillEnd.toISOString(),
+          let earliest = Date.now();
+          let latest = 0;
+
+          Object.values(generatedGapEvents).forEach(({ gapEvents }) => {
+            gapEvents
+              .flatMap((event) => event.unfilled_intervals)
+              .forEach(({ gte, lte }) => {
+                earliest = Math.min(earliest, new Date(gte).getTime());
+                latest = Math.max(latest, new Date(lte).getTime());
+              });
+          });
+
+          backfillEnd = new Date(latest);
+          backfillStart = new Date(earliest);
+        });
```

Comment:
nit: might be worth putting a small comment describing what you're setting up here, without context it's a bit tough to glean